hudi-spark-datasource/hudi-spark/src/main/scala/org/apache/hudi/HoodieSparkSqlWriter.scala

```diff
@@ -159,7 +159,10 @@ object HoodieSparkSqlWriter {
 
           // Convert to RDD[HoodieRecord]
           val genericRecords: RDD[GenericRecord] = HoodieSparkUtils.createRdd(df, schema, structName, nameSpace)
-          val shouldCombine = parameters(INSERT_DROP_DUPS_OPT_KEY.key()).toBoolean || operation.equals(WriteOperationType.UPSERT);
+          val shouldCombine = parameters(INSERT_DROP_DUPS_OPT_KEY.key()).toBoolean ||
+            operation.equals(WriteOperationType.UPSERT) ||
+            parameters.getOrElse(HoodieWriteConfig.COMBINE_BEFORE_INSERT_PROP.key(),
```

Comment:
here is a bug?

Reply:
Yes,  if we have enable the COMBINE_BEFORE_INSERT_PROP for insert, it has not compute the pre combine field value which will result incorrect result for insert with duplicate records.

Reply:
If the `COMBINE_BEFORE_INSERT_PROP` has enabled, `SparkInsertCommitActionExecutor` will do the  `combineOnCondition` in `AbstractWriteHelper`  which will need the `precombine` value .
 But here we have ignored the case of `COMBINE_BEFORE_INSERT_PROP` which will not extract the `precombine` value to the HoodiePayload.

Reply:
I digged in to understand this. Actually INSERT_DROP_DUPS_OPT_KEY is used at the Dataource layer and COMBINE_BEFORE_INSERT_PROP is used in writeClient layer.  In other words, both are referring to same config only. [DataSourceUtils](https://github.com/apache/hudi/blob/6353fc865f43854e0e185af33e8ad091c8870d78/hudi-spark-datasource/hudi-spark-common/src/main/java/org/apache/hudi/DataSourceUtils.java#L178) converts INSERT_DROP_DUPS_OPT_KEY's value as COMBINE_BEFORE_INSERT_PROP. 
So, don't think we need to check both INSERT_DROP_DUPS_OPT_KEY and COMBINE_BEFORE_INSERT_PROP.

Reply:
@pengzhiwei2018 do you agree with siva's analysis above?

Reply:
@vinothchandar  well I think  INSERT_DROP_DUPS_OPT_KEY is some different from COMBINE_BEFORE_INSERT_PROP. 
**INSERT_DROP_DUPS_OPT_KEY**:  is used to drop the duplicate record in the target table.
`COMBINE_BEFORE_INSERT_PROP`: is used to combine the duplicate record in the input.
So they are not total the same config. IMO.